detailedinfo/domainmessageflow.md

```diff
@@ -0,0 +1,108 @@
+  - The order in which the message occurs in the flow being modelled. The order is the position and the sequence of the message in the message flow. The order indicates when and how the message is sent and received. The order can be expressed using numbers, letters, or symbols. For example, you could use numbers to indicate the chronological order of the messages, such as 1, 2, 3, etc. For example, the place order command could be the first message in the message flow, so it could have the order 1.
+
+After you have created the diagrams for each scenario, you can review them and look for any gaps, inconsistencies, or errors. You can also compare them with the requirements and the expectations of the stakeholders. You can use the following questions to guide your review:
+
+- Are the bounded contexts clearly defined and named?
+- Are the messages clear, consistent, and complete?
+- Are the message types appropriate for the purpose and the direction of the message?
+- Are the message data minimal, sufficient, and structured?
+- Are the message orders correct and logical?
+- Are the message flows complete and coherent?
+- Do the message flows cover all the scenarios and outcomes?
+- Do the message flows meet the requirements and the expectations of the stakeholders?
+
+If you find any issues or areas for improvement, you can revise your diagrams and update your message flows accordingly. You can also get feedback from other developers, architects, or domain experts to validate and refine your message flows. By doing this, you can ensure that your message flows are accurate, consistent, and comprehensive. With this iterative approach you can also verify if your EventStorming has been valid. You can take the newly gained insights to iterate on your EventStorming.
+
+## Anti-Pattern
+
+An anti-pattern is a common but ineffective or harmful solution to a problem, that has more negative consequences than positive ones. It is the opposite of a pattern, which is a good and proven solution to a problem. Anti-patterns can be found in various domains, such as software engineering, project management, and business processes. The following are anti-pattern we have identified and observed in several workshops. They are often indicators that the domain is not broken down into independent bounded contexts. By not addressing them you might inflict unwanted and unneeded dependencies into your application design.
+
+### Synchronous Dependencies / Communication
```

Comment:
I think this paragraph overshoots the mark a little and needs a bit more nuance. 

First, I think it's important to differentiate between synchronous requests at a technical and at a process level. I do agree that when a process has to wait for a synchronous response from some other service in order to do it's job and the unavailability of said component leads to an error that is observable by clients, that is a problem and can be solved by designing the process asynchronously.

However, such design can be achieved with synchronous means like HTTP feeds or transactional outbox. So saying "synchronous = bad" is too coarse and I think can lead to wrong conclusions like "okay I need Kafka/rabbitmq/whatever everywhere".

Reply:
Changed it